### PR TITLE
[AAE-7154] Automate upload file dialog

### PR DESCRIPTION
**Pull request by Qetikv**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [x] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

https://alfresco.atlassian.net/browse/AAE-7154

**What is the new behaviour?**

e2e test is shared

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: projects/aca-testing-shared/src/pages/page.ts [hunk 1]
```diff
@@ -82,13 +83,11 @@ export abstract class Page {
...
   async isUploadDialogOpen(): Promise<boolean> {
-    return isPresentAndDisplayed(this.uploadDialog);
+    return this.uploadFilesDialog.isUploadDialogOpen();
   }
 
   async closeUploadDialog(): Promise<void> {
-    if (await this.isUploadDialogOpen()) {
-      await BrowserActions.click(this.closeUploadButton);
-    }
+    await this.uploadFilesDialog.closeUploadDialog();
```
Comment (MichalFidor) [inline projects/aca-testing-shared/src/pages/page.ts:90]: You can remove this one line method and call it directly in spec file

File: projects/aca-testing-shared/src/pages/page.ts [hunk 2]
```diff
@@ -82,13 +83,11 @@ export abstract class Page {
   }
 
   async isUploadDialogOpen(): Promise<boolean> {
-    return isPresentAndDisplayed(this.uploadDialog);
+    return this.uploadFilesDialog.isUploadDialogOpen();
```
Comment (MichalFidor) [inline projects/aca-testing-shared/src/pages/page.ts:86]: You can remove this one line method and call it directly in spec file

File: projects/aca-testing-shared/src/components/dialog/upload-files-dialog.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+export class UploadFilesDialog {
+  uploadDialog = TestElement.byCss('.adf-upload-dialog');
+  closeUploadButton = TestElement.byCss('.adf-upload-dialog [id="adf-upload-dialog-close"]');
+  minimizeButton = TestElement.byCss('.adf-upload-dialog mat-icon[title="Minimize"]');
+  maximizeButton = TestElement.byCss('.adf-upload-dialog mat-icon[title="Maximize"]');
+  uploadedFiles = TestElement.byCss('.adf-file-uploading-row__name');
+
+  async isUploadDialogOpen(): Promise<boolean> {
+    try {
+      await this.uploadDialog.isVisible();
```
Comment (MichalFidor) [inline projects/aca-testing-shared/src/components/dialog/upload-files-dialog.ts:37]: TestElement.isVisible() already has the try/catch logic so you don't need to do this logic here. Just use `await this.uploadDialog.isVisible();` in spec file and it'll be sufficient

File: e2e/suites/actions/upload-download/upload-file.test.ts [hunk 4]
```diff
@@ -50,19 +50,73 @@ describe('Upload files', () => {
...
+
+    await page.clickRecentFiles();
+    await expect(page.uploadFilesDialog.uploadDialog.isVisible()).toBe(true);
+
+    await page.clickFavorites();
+    await expect(page.uploadFilesDialog.uploadDialog.isVisible()).toBe(true);
 
-    expect(await dataTable.isItemPresent('upload-file.test.ts')).toBe(true, 'file not uploaded');
+    await page.clickTrash();
+    await expect(page.uploadFilesDialog.uploadDialog.isVisible()).toBe(true);
```
Comment (MichalFidor) [inline e2e/suites/actions/upload-download/upload-file.test.ts:120]: I think, that w can check this thing in approximately 2 different cases. You don't need to check it by clicking in each link and expecting that dialog is still visible. Please change this code and leave only this:
```
await expect(page.uploadFilesDialog.uploadDialog.isVisible()).toBe(true);
await page.clickPersonalFiles();
await expect(page.uploadFilesDialog.uploadDialog.isVisible()).toBe(true);
```

File: e2e/suites/actions/upload-download/upload-file.test.ts [hunk 5]
```diff
@@ -50,19 +50,73 @@ describe('Upload files', () => {
...
+    await page.uploadFilesDialog.minimizeButton.click();
+    await expect(await page.uploadFilesDialog.uploadedFiles.waitNotVisible()).toBe(true);
+
+    await page.uploadFilesDialog.maximizeButton.click();
+    await expect(await page.uploadFilesDialog.uploadedFiles.waitVisible()).toBe(true);
+  });
+
+  describe('[T14752053]', () => {
+    it('Upload history is expunged on browser refresh ', async () => {
+      await page.refresh();
```
Comment (MichalFidor) [inline e2e/suites/actions/upload-download/upload-file.test.ts:86]: As I remember the refresh thing is something we want to get rid of in all of E2Es (there is also a PR by @eromano created in alfresco-apps)

**Review by MichalFidor — CHANGES_REQUESTED** [review]

File: projects/aca-testing-shared/src/components/dialog/upload-files-dialog.ts (hunk 3 above)
Comment (Qetikv) [reply projects/aca-testing-shared/src/components/dialog/upload-files-dialog.ts]: Fixed

File: projects/aca-testing-shared/src/pages/page.ts (hunk 2 above)
Comment (Qetikv) [reply projects/aca-testing-shared/src/pages/page.ts]: Fixed

File: projects/aca-testing-shared/src/pages/page.ts (hunk 1 above)
Comment (Qetikv) [reply projects/aca-testing-shared/src/pages/page.ts:86]: closeUploadDialog() is also used in other spec file that I was not working on. Should I still remove it and change the other test file as well?

File: e2e/suites/actions/upload-download/upload-file.test.ts (hunk 4 above)
Comment (Qetikv) [reply e2e/suites/actions/upload-download/upload-file.test.ts]: I have changed it. I was following the manual regression test case and it was checked from these places

File: e2e/suites/actions/upload-download/upload-file.test.ts (hunk 5 above)
Comment (Qetikv) [reply e2e/suites/actions/upload-download/upload-file.test.ts:86]: In the test description it is said that after page refresh the dialog shouldn't be present,  should I delete this e2e test or is there any other way to test it?

File: projects/aca-testing-shared/src/pages/page.ts (hunk 1 above)
Comment (MichalFidor) [reply projects/aca-testing-shared/src/pages/page.ts:86]: Yes, please :)

File: e2e/suites/actions/upload-download/upload-file.test.ts (hunk 5 above)
Comment (MichalFidor) [reply e2e/suites/actions/upload-download/upload-file.test.ts:86]: https://github.com/Alfresco/alfresco-apps/pull/3035#issuecomment-971785029

Yes, please, remove it.

Comment (Qetikv) [reply e2e/suites/actions/upload-download/upload-file.test.ts]: Done

File: projects/aca-testing-shared/src/pages/page.ts (hunk 1 above)
Comment (Qetikv) [reply projects/aca-testing-shared/src/pages/page.ts]: Done

**Review by MichalFidor — APPROVED** [review]

**Pull request merged**